Objects/longobject.c

```diff
@@ -4309,18 +4315,34 @@ long_rshift(PyLongObject *a, PyLongObject *b)
     }
     else {
         shiftby = PyLong_AsSsize_t((PyObject *)b);
-        if (shiftby == -1L && PyErr_Occurred())
-            return NULL;
-        if (shiftby < 0) {
-            PyErr_SetString(PyExc_ValueError,
-                            "negative shift count");
-            return NULL;
+        if (shiftby >= 0) {
+            wordshift = shiftby / PyLong_SHIFT;
+            loshift = shiftby - wordshift * PyLong_SHIFT;
```

Comment:
Left shift uses such code. I suppose this is because multiplication and substraction are faster than the "%" operation. If there is a reason to use it for left shit, there is a reason to use it for right shift.

The new code for left and right shifts are almost exact duplicates (there is additional optimization for right shift, but it is not very important and can be omitted). Is it worth to extract it in the helper function?